### PR TITLE
Front-end code standard pass

### LINKED ISSUE #259: Front end code standard pass & clean up
## Description

Remove useless code and refactor/clean up code in the front end folder

## Possible Implementation

Go through the front-end folder and clean up code

### LINKED ISSUE #268: Add Edit/Delete DM+Character Entry options from the Character Page
## Description
The character page should have buttons which allow the user to :
- Edit a DM Entry
- Delete a DM Entry
- Edit a Character Entry
- Delete a Character Entry

### LINKED ISSUE #272: Font at the pagination of the DMEntryTable for rows per page dropdown is wrong
## Description

Wrong font for the number of rows per page select field in the pagination of the DMEntryTable.

## Expected Behavior

It should be the right font.

## Actual Behavior

It is not the right font. 

## Steps to Reproduce

1. Go to characters tab, click on DM ENTRY button.
2. Look at the rows per page select field.
3. 

## Context

UI affected. UI not consistent with other table. Also it's ugly.

## Possible Fix

# Other

### PR DESCRIPTION
**Pull request by willyyhuang**
Closes #259 
Closes #272 
Closes #268 

Changes
- Added a select component
- Added option constant
- Refactor DM entry and entry form
- Added path for `@Utils`
- Added a divider in drawer
- Added edit DM entry/entry form drawer
- Added a new hook for edit drawer
- Fix dm entry table theme provider

### CODE REVIEW

File: resources/js/Components/Form/EntryCreateForm.tsx [hunk 1]
```diff
@@ -119,196 +115,143 @@ const EntryCreateForm = ({
...
-                        select
-                        required
-                        label='Adventure Title'
-                        name='Adventure Title'
-                        SelectProps={{
-                            MenuProps: {PaperProps: {sx: {maxHeight: 300}}},
+        <Grid container spacing={2}>
+            <Grid item xs={12}>
+                <Typography>
+                    Fill out the following fields with your DM Entry details.
```
Comment (AnnieTran13) [inline resources/js/Components/Form/EntryCreateForm.tsx:121]: This should be "Fill out the following fields with your entry details." and not DM entry details

**Review by pascwhale — COMMENTED** [review]
I love the divider in the forms!

For some reason I wasn't able to create a DM Entry. I got the following error: `Undefined array key "dungeon_master"`. Editing an entry also didn't work when the page was updated.

File: resources/js/Components/Form/EntryCreateForm.tsx (hunk 1 above)
Comment (willyyhuang) [reply resources/js/Components/Form/EntryCreateForm.tsx:121]: will fix this, thanks for pointing out

**Comment by willyyhuang** [comment]
> I love the divider in the forms!
> 
> For some reason I wasn't able to create a DM Entry. I got the following error: `Undefined array key "dungeon_master"`. Editing an entry also didn't work when the page was updated.

@m-triassi is having some issues with editing, we will see what we can do

**Review by pascwhale — COMMENTED** [review]
The Entry Form hint is now correct. However, editing an entry gives me a constraint error. Refreshing the pages applies the change however.

**Review by pascwhale — APPROVED** [review]
No errors on the frontend side of things, I was able to create a DM Entry without any errors.

**Pull request merged**